tests/ui/parser/f16-f128.rs

```diff
@@ -1,37 +0,0 @@
-// Make sure we don't ICE while incrementally adding f16 and f128 support
-
-mod f16_checks {
-    const A: f16 = 10.0; //~ ERROR cannot find type `f16` in this scope
```

Comment:
Your code handles this correctly, but can you add a new pass test with a custom struct called f16?

Reply:
This comment is in the sanity check from https://github.com/rust-lang/rust/pull/121728#discussion_r1505689228 which I removed here, but I do have that as part of the feature gate test https://github.com/rust-lang/rust/pull/121926/files/a7ce0b8fc666b64506cb7677706b3cfd9cf87cf6#diff-17cba0705a57a125f2d1eb3ddd12cee1ab97546c322cced8fcedec40a370f1a5R21-R34. Is there a better place for that?

Reply:
I put these in a different file so they'll stick around after the feature gate, https://github.com/rust-lang/rust/pull/121926/files#diff-139f82db73c1edb30cdb6cd881806233774bad18de0462af6b1e8bf1cef019e7